### PR TITLE
 Update github action setting environment variable workflow command

### PR DESCRIPTION
**Pull request by khs1994**
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable

### CODE REVIEW

**Comment by caspervonb** [comment]
What is the purpose of this patch?

**Comment by khs1994** [comment]
https://github.com/actions/toolkit/security/advisories/GHSA-mfwh-5m23-j46w

old command will deprecated

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -100,7 +100,8 @@ jobs:
         # that are needed during the build process. Additionally, this works
         # around a bug in the 'cache' action that causes directories outside of
         # the workspace dir to be saved/restored incorrectly.
-        run: echo "::set-env name=CARGO_HOME::$(pwd)/.cargo_home"
+        shell: pwsh
```
Comment (caspervonb) [inline .github/workflows/ci.yml:103]: Why change the shell from default?

Comment (khs1994) [reply .github/workflows/ci.yml:103]: linux and windows diff

if use bash in windows `CARGO_HOME: /d/a/deno/deno/.cargo_home`

Comment (khs1994) [reply .github/workflows/ci.yml:103]: `pwsh` exists in all os

**Comment by khs1994** [comment]
ping @ry @fd

**Comment by khs1994** [comment]
#8020

**Pull request closed without merging**